docs/configs-api.md

```diff
@@ -0,0 +1,95 @@
+# Configs API
+
+The configs api provides API-driven multi-tenant approach to prometheus' Rule files and Alertmanager configs.
+
+Each tenant has a single Rule file and Alertmanager Configuration. A POST operation will effectively replace the existing copy with the configs provided in the request body.
+
+Although Prometheus works with YAML files by default, these HTTP API's only accept JSON. You may convert your YAML to JSON using a tool such as [yaml2json](https://github.com/bronze1man/yaml2json).
+
+### Manage Rules
+
+`GET /api/prom/configs/rules` - Get current rule file
+
+`POST /api/prom/configs/rules` - Replace current rule file
+
+Returns and Accepts a JSON object representing a [Prometheus rule file](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/#recording-rules). 
+
+Example:
```

Comment:
It's been quite awhile since I used this api, but I though these were wrapped with some extra data (rule file version, then a map of all the rule files). I think the relevant struct is [`configs.RulesConfig`](https://github.com/cortexproject/cortex/blob/master/pkg/configs/configs.go#L139).